ci-operator/step-registry/cucushift/installer/check/ibmcloud/resourcegroup/cucushift-installer-check-ibmcloud-resourcegroup-commands.sh

```diff
@@ -0,0 +1,96 @@
+function getResources() {
+    "${IBMCLOUD_CLI}" resource service-instances --type all --output JSON | jq -r '.[]|.name+" "+.resource_id' | sort
+}
+
+function checkResourcesInVPC() {
+    local preResourceFile provisioned_rg_file resource_group curResourceFile cmpRst
+    resource_group=$1
+    preResourceFile="${SHARED_DIR}/vpc_resources"
+    "${IBMCLOUD_CLI}" target -g ${resource_group}
+    curResourceFile=$(mktemp)
+    getResources > ${curResourceFile}
+    echo "diff ${curResourceFile} ${preResourceFile} ..."
+    cmpRst=$(diff ${curResourceFile} ${preResourceFile})
+    if [ -n "$cmpRst" ]; then
+        echo "ERROR: resources differ: ${cmpRst}"
+        return 1
+    else
+        echo "checkResourcesinVPC PASS"  
+        return 0
+    fi
+}
+
+if [ -f "${SHARED_DIR}/kubeconfig" ] ; then
+    export KUBECONFIG=${SHARED_DIR}/kubeconfig
+else
+    echo "ERROR: fail to get the kubeconfig file under ${SHARED_DIR}!!"
+    exit 1
+fi
+
+if [ -f "${SHARED_DIR}/proxy-conf.sh" ] ; then
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+check_result=0
+
+provisioned_rg_file="${SHARED_DIR}/ibmcloud_resource_group"
+cluster_resource_group=$(jq -r .ibmcloud.resourceGroupName ${SHARED_DIR}/metadata.json)
+if [ -f "${SHARED_DIR}/ibmcloud_resource_group" ]; then
+    vpc_resource_group=$(cat "${provisioned_rg_file}")
+    rg_file="${SHARED_DIR}/ibmcloud_cluster_resource_group"
+    ibmcloud_login
+    if [ "${CREATE_CLUSTER_RESOURCE_GROUP}" == "yes" ]; then
+        if [ -s "${rg_file}" ]; then            
+            if [ "${vpc_resource_group}" == "${cluster_resource_group}" ]; then
+                echo "ERROR: vpc and cluster use the same resource group ${cluster_resource_group} when CREATE_CLUSTER_RESOURCE_GROUP is set!"
+                check_result=1
+            fi
+            rgName=$(cat "${rg_file}")
+            if [ "${rgName}" != "${cluster_resource_group}" ]; then
+                echo "ERROR: in ${rg_file} saved ${rgName} is not same with the cluster name ${cluster_resource_group}!"
+                check_result=1
+            fi
+
+            if ! checkResourcesInVPC ${vpc_resource_group}; then
+                check_result=1
+            fi
+        else
+            echo "ERROR: fail to find ${rg_file}"
+            check_result=1
+        fi
+    else
+        if [ -s "${rg_file}" ]; then
+            echo "ERROR: find the ${rg_file} when CREATE_CLUSTER_RESOURCE_GROUP is not YES!"
+            check_result=1
+        fi
+    fi
+fi
+
+#check the cluster in the cluster resource group
+nodes_count=$(oc get nodes --no-headers | wc -l)
+clusterName=$(jq -r .clusterName ${SHARED_DIR}/metadata.json)
+insCount=$(${IBMCLOUD_CLI} is ins --resource-group-name ${cluster_resource_group} --output JSON | jq '.|length')
+clusterInsCount=$(${IBMCLOUD_CLI} is ins --all-resource-groups | grep ${clusterName} -c)
+if [ "${nodes_count}" -ne "${insCount}" ] || [ "$insCount" -ne "$clusterInsCount" ]; then
```

Comment:
From `[ "${nodes_count}" -ne "${insCount}" ]` condition, we already checked all the cluster nodes are placed into the cluster resource group, I think that is enough, do not need to run the `clusterInsCount` check.

Reply:
I kept clusterInsCount checking, no more cluster instances in other groups.

Reply:
Think about that, `nodes_count` is the number of all the cluster nodes, which is validated to ensure be equal to `nodes_count` which is the number of vm in the `cluster_resource_group`, so that means it is impossible to some nodes are existing out of `cluster_resource_group` with `insCount` == `nodes_count` precondition.